src/Codegen/Codegen.php

```diff
@@ -53,6 +53,7 @@ final class Codegen {
       ?'root_schema' => TSchema,
       ?'root_schema_path' => string,
     ),
+    ?'discard_additional_properties' => bool,
```

Comment:
Is the intent that we would only specify this for top level API validation? Sometimes I think we want to be strict to avoid people providing whatever, us ignoring it, and then wanting to use the property they're providing and it becoming a "backwards incompatible" change.